scripts/check_compliance.py

```diff
@@ -588,6 +588,118 @@ def run(self):
+               "E0710,"  # Raising non-BaseException
+               "E0711,"  # NotImplemented instead of NotImplementedError
+               "E0712,"  # Catching non-BaseException
+               "E1003,"  # Bad first argument to super()
+               "E1004,"  # Missing argument to super()
+               #"E1101,"  # Non-existent object member
+               "E1102,"  # Not callable
+               #"E1103,"  # Non-existent object member, more complex case (?)
+               "E1111,"  # Assigning to function call that doesn't return
+               "E1122,"  # Missing argument in function call
+               #"E1123,"  # Bad keyword argument in function call
+               "E1124,"  # Argument passed both positionally and as keyword
+               "E1200,"  # Unsupported logging format character
+               "E1201,"  # Logging format string ends in middle of conversion specifier
+               "E1205,"  # Too many arguments for logging format string
+               "E1206,"  # Not enough arguments for logging format string
+               "E1300,"  # Unsupported format character
+               "E1301,"  # Format string ends in middle of conversion specifier
+               "E1302,"  # Mixed named and unnamed conversion specifiers
+               "E1303,"  # Expected mapping for format string, not %s
+               "E1304,"  # Missing key in format string dictionary
+               "E1305,"  # Too many arguments for format string
+               "E1306,"  # Not enough arguments for format string
+               "E1310,"  # Suspicious argument in call
+               "R1711,"  # Redundant 'return'
+               "W0105,"  # String statement has no effect
+               "W0611,"  # Unused import
+               "W0612,"  # Unused variable
+               "W0301,"  # Unnecessary semicolon
+               "W0311,"  # Bad indentation
+               "W1401,"  # Backslash in non-r string
+              )
+
+        py_files = self.py_files()
+        try:
+            # Work around a crash in pylint 2.2.2:
+            # https://github.com/PyCQA/pylint/issues/2906
+            py_files.remove("boards/xtensa/intel_s1000_crb/support/create_board_img.py")
+        except ValueError:
+            self.error("Workaround in check_compliance.py no longer needed? "
+                       "Please check (grep for this message).")
+
+        process = subprocess.Popen(cmd.split() + py_files,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE,
+                                   cwd=ZEPHYR_BASE)
+
+        stdout, stderr = process.communicate()
+        if process.returncode or stderr:
+            # Issues found, or a problem with pylint itself
+            self.add_failure(stdout.decode("utf-8") + stderr.decode("utf-8"))
+
+
+    def py_files(self):
+        """
+        Returns a list with the paths of all .py files, relative to the Zephyr
```

Comment:
Running on unmodified files is a waste of time, the purpose of this script is to check incremental commits. Other check(s) use `git diff --name-only`, it should be possible to refactor and re-use.

Reply:
It's very simple, rock solid, and fast enough.

Reply:
> It's very simple, rock solid

The `git diff --name-only` command has already been "in production" for months so it's rock solid too and barely longer. It also reflects the general purpose of this script as documented in its `--commits` argument help which this check silently drops on the floor.

> and fast enough

check_compliance.py -m pylint lasts 7 seconds and uses 52 seconds of CPU time on my workstation for a zero python change commit so: no, it's not "fast enough". @nashif may also have an opinion about the Shippable bill.

> My modus operandi is to clean up the entire codebase first

Great, once that's done you don't need to have every Shippable instance repeatedly and pointlessly scan the same, unchanged python code over and over.  If the config file is extracted from the code then anyone who wants to scan the whole codebase again [*] can at anytime, they don't even need to run this script.

[*] for instance to evolve the config incrementally without requiring a "big bang" fix.

Reply:
Alright, I'll change it. It usually means you end up with failures getting in through merges and CI overrides and stuff over time, so I think it's a bad idea though.

Reply:
The nightly builds are the best place to catch these, so they don't confuse and bother submitters with failures totally unrelated to their submission. Assuming a separate config file it should be as simple as adding literally one line in the nightly builds.

Reply:
Now runs `pylint` on just new/modified Python scripts.

Reply:
@marc-hb 
Refresh ;)

Reply:
Or do you still want it in the nightly builds?

The Python files are known to be warning-free now, so it shouldn't be possible to get false positives, unless something slips through through a forced merge or the like.